### PR TITLE
[11.0][MIG] product_by_supplier

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
As @lreficent commented here https://github.com/OCA/purchase-workflow/pull/596#issuecomment-434623285, I made the migration of `product_by_supplier`.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

File: addons/purchase/migrations/11.0.1.2/pre-migration.py [hunk 1]
```diff
@@ -6,6 +6,24 @@
 from openupgradelib import openupgrade
 
 
+def detect_product_by_supplier(cr):
+    cr.execute(
+        """
+        SELECT id
+        FROM ir_model_data
+        WHERE module = 'product_by_supplier' AND
```
Comment (pedrobaeza) [inline addons/purchase/migrations/11.0.1.2/pre-migration.py]: This is going to be always Falsy, as you have merged module in `purchase`. Any way, do this or other check in post for avoiding this ugly hack of creating a fake column. Do you know that in OpenUpgrade views are not removed? Use that.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: addons/purchase/migrations/11.0.1.2/pre-migration.py (hunk 1 above)
Comment (StefanRijnhart) [reply addons/purchase/migrations/11.0.1.2/pre-migration.py]: For the record, while OpenUpgrade does not drop seemingly obsolete columns and tables like Odoo does, it does delete module data (such as views) that is not defined as noupdate but only once all the modules have been processed.

Comment (pedrobaeza) [reply addons/purchase/migrations/11.0.1.2/pre-migration.py]: Thanks for the complementary information, Stefan!

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**